packages/mobile/components/SidebarArticle.tsx

```diff
@@ -44,8 +46,20 @@ const SidebarArticle = ({
     reload && refetch();
   }, [reload]);
 
+  const toggleFavouriteAction = () =>
+    toggleFavouriteMutation({
+      variables: {
+        articleId: article.id,
+      },
+    })
+      .then(({ data: { toggleFavourite } }) => {
+        setFavourite(toggleFavourite);
+      })
+      .catch((err) => console.log(`Error togglefavourite: ${err}`));
+
   const article = data?.article;
   const titleId = `${article?.title}-${article?.id}`;
+  isFavourite === null && article && setFavourite(article?.favourited);
```

Comment:
Might this "=== null" be changed to "!variable" or is it necessary to check if null?

Reply:
The variable is only null before the article is loaded, then it's a boolean so checking for `!variable` wont do

Reply:
Excellent